### DISCUSSION TITLE
Flickering

### DISCUSSION THREAD

**Discussion by zensimilia** [lovyan03/LovyanGFX#363] [Q&A]
Hello!

Why clearing screen every frame straight on `LGFX_Device` object causes the screen to flickering? But if use sprite over screen and clear sprite then push it, this works like a charm. An extra sprite to fit the screen requires more memory.

CVBS panel in my case.

**Comment by tobozo** [comment]
clearScreen() is actually fillScreen( black ) so the flickering is to be expected

if you have enough spare memory you can eventually use LGFX_Sprite and sprite.pushSprite() to prevent that, but there are limitations (e.g. max 230x230@16bpp)

other methods are blit, partial update, framebuffer

**Reply by zensimilia** [reply]
> clearScreen() is actually fillScreen( black ) so the flickering is to be expected

I don't fully understand why this is not expected from the `LGFX_Sprite` object in this case. It is also filled with black every frame.

> other methods are blit, partial update, framebuffer

Maybe there are some examples for each method?

**Reply by zensimilia** [reply]
Should I prepare a framebuffer (the size of the screen) and instead of filling the screen, draw a new framebuffer on top? 

Questions is: this is the same as use `LGFX_Sprite`?

**Reply by tobozo** [reply]
LGFX_Sprite has a framebuffer, I'm not expert enough to explain why, but I know it doesn't flicker when you use it.

You can use sprite.clearScreen(), then redraw your 3D model using `sprite.draw*` methods instead of `tft.draw*`, then call sprite.pushSprite(), and the flicker should be gone.

If you really want a 320x240 sprite, I recommend you set an 8 bits colors palette using  `sprite.setColorDepth(8)` prior to calling `sprite.createSprite()`, the gradients won't look nice though.

**Reply by zensimilia** [reply]
This is exactly what I do, but I'm starting to worry about the size of the memory used when I use the framebuffer fit the screen. At the same time, the `LGFX_Device` object also have framebuffer and consumes memory, but is not actually used for drawing. Correct me if I'm wrong.

Where i can find more info about to use blit and partial update methods?

**Reply by tobozo** [reply]
> blit and partial update methods

one is a subset of the other

https://en.wikipedia.org/wiki/Bit_blit

The CVBS framebuffer can be used as the 'blit' background image (it should be called a blit sprite so it's kinda confusing), and the LGFX_Sprite can be used as the mask generator.
